### ISSUE TITLE
Visible chat history length

### ISSUE THREAD

**Issue by burtmaplewud** [BoltAI/BoltAI#130]
(Setapp v1.16.11) My chats show only a subset of the history, with a rolling cutoff that removes earlier messages. The entire history is available via export, however. This happens regardless of the context setting. Bug? Intended?

**Comment by longseespace** [comment]
Yeah, I've heard of another report on this. It's likely a bug but the behavior was intended.

For better performance, BoltAI only loads 10 message initially. And when you scroll up, it _should_ load more messages. Apparently, it doesn't seem to behave correctly and will just stop loading.

I will work on the fix soon.

**Comment by longseespace** [comment]
Did you still encounter this issue @burtmaplewud ?
